src/cmd/windows/durable_task_monitor.go

```diff
@@ -0,0 +1,160 @@
+		sysAttr.CmdLine = executable + " " + args
+	} else {
+		// Delimiter is a comma WITH a space to handle complex arguments that require commas. For example, an argument that includes a method
+		// call with args separated by a comma. The method args can be separated by commas with NO space so they do not accidentally get split
+		scriptCmd = exec.Command(executable, strings.Split(args, ", ")...)
+	}
+	scriptCmd.SysProcAttr = &sysAttr
+
+	if outputPath != "" {
+		// capturing output
+		outputFile, err := os.Create(outputPath)
+		if err != nil {
+			launchLogger.Println(err.Error())
+			common.RecordExit(-2, resultPath, launchLogger)
+			return
+		}
+		defer outputFile.Close()
+
+		// Note: Go writes the output in utf8 WITHOUT a bom. No need for any encoding conversions
+		scriptCmd.Stdout = outputFile
+		scriptCmd.Stderr = scriptLogger.Writer()
+	} else {
+		scriptCmd.Stdout = scriptLogger.Writer()
+		scriptCmd.Stderr = scriptCmd.Stdout
+	}
+
+	for i := 0; i < len(scriptCmd.Args); i++ {
+		launchLogger.Printf("args %v: %v\n", i, scriptCmd.Args[i])
+	}
+	err := scriptCmd.Start()
+	if common.CheckIfErr(scriptLogger, err) {
+		common.RecordExit(-2, resultPath, scriptLogger)
+		return
+	}
+	pid := scriptCmd.Process.Pid
+	launchLogger.Printf("launched %v\n", pid)
+	err = scriptCmd.Wait()
+	common.CheckIfErr(scriptLogger, err)
+	resultVal := scriptCmd.ProcessState.ExitCode()
+	launchLogger.Printf("script exit code: %v\n", resultVal)
+
+	common.RecordExit(resultVal, resultPath, launchLogger)
+}
+
+func main() {
+	var controlDir, resultPath, logPath, executable, args, outputPath string
+	var debug, daemon bool
+	const controlFlag = "controldir"
+	const resultFlag = "result"
+	const logFlag = "log"
+	const executableFlag = "executable"
+	const argsFlag = "args"
+	const outputFlag = "output"
+	const debugFlag = "debug"
+	const daemonFlag = "daemon"
+	flag.StringVar(&controlDir, controlFlag, "", "working directory")
+	flag.StringVar(&resultPath, resultFlag, "", "full path of the result file")
+	flag.StringVar(&logPath, logFlag, "", "full path of the log file")
+	flag.StringVar(&executable, executableFlag, "", "path to the executable being launched")
+	flag.StringVar(&args, argsFlag, "", "(optional) argument(s) to the executable, separated by commas WITH spaces")
+	flag.StringVar(&outputPath, outputFlag, "", "(optional) if recording output, full path of the output file")
+	flag.BoolVar(&debug, debugFlag, false, "(optional) noisy output to log")
+	flag.BoolVar(&daemon, daemonFlag, false, "(optional) Free binary from parent process")
+	flag.Parse()
+
+	// Validate that the required flags were all command-line defined
+	required := []string{controlFlag, resultFlag, logFlag, executableFlag}
+	defined := make(map[string]string)
+	flag.Visit(func(f *flag.Flag) {
+		defined[f.Name] = f.Value.String()
+	})
+	if !common.ValidateFlags(defined, required) {
+		os.Exit(-2)
+	}
+
+	fmt.Fprintf(os.Stdout, "Parent pid is: %v\n", os.Getppid())
```

Comment:
Will this show up in the build log in Jenkins?

Reply:
Only when debug is enabled. `durable-task` will enable it with `LAUNCH_DIAGNOSTICS`

Reply:
yes it will. Removed in e87b44bb9fc6c71cebd5be23ec0a757889c35423